app/models/user.rb

```diff
@@ -8,4 +9,35 @@ class User < ApplicationRecord
   validates :password, presence: true, length: { minimum: Settings.validation.password_len_min }
 
   has_secure_password
+
+  # Returns the hash digest of the given string.
+  def self.digest string
+    if ActiveModel::SecurePassword.min_cost
+      cost = BCrypt::Engine::MIN_COST
+    else
+      cost = BCrypt::Engine.cost
+    end
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  # Returns a random token.
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
```

Comment:
khi define nhiều hơn 2 class methods thì em dùng cú pháp như sau nhé
```suggestion
class << self
  def digest string
    if ActiveModel::SecurePassword.min_cost
      cost = BCrypt::Engine::MIN_COST
    else
      cost = BCrypt::Engine.cost
    end
    BCrypt::Password.create(string, cost: cost)
  end

  # Returns a random token.
  def new_token
    SecureRandom.urlsafe_base64
  end
end
```